### PR TITLE
Pattern Library additions

### PR DESCRIPTION
**Pull request by shiftsave**
This PR implements some base components and their styling, list to follow.

### CODE REVIEW

File: src/routes/patterns/components/Alerts.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+import Card from 'components/Card';
+
+export default class Alerts extends Component {
+  state = {
+    isDefault: false,
+    isDanger: false
+  };
+
+  showDefault = () => {
```
Comment (hudakdidit) [inline src/routes/patterns/components/Alerts.js]: Why not just `() {`

Comment (jonathanhudak) [inline src/routes/patterns/components/Alerts.js:11]: Why not just `() {`

File: src/routes/patterns/components/Alerts.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      isDanger: false
+    });
+  };
+
+  render() {
+    const { isDefault, isDanger } = this.state;
+
+    return (
+      <Card>
+        {isDefault &&
```
Comment (jonathanhudak) [inline src/routes/patterns/components/Alerts.js:37]: There is a fair amount of duplication in each of these cases. I would suggest refactoring to simplify. You could utilize state-less components that just take props to help reduce the repetition. Happy to help if needed.

File: src/routes/patterns/components/Alerts.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              <button>Process file</button>
+            </div>
+          </div>}
+
+        {isDanger &&
+          <div danger>
+            <div className='message'>
+              <p>
+                <strong>Upload failed!</strong>
+                {' '}
```
Comment (jonathanhudak) [inline src/routes/patterns/components/Alerts.js:59]: `&nbsp;`?

File: src/styles/global/mixins.styl [hunk 4]
```diff
@@ -6,10 +6,37 @@
...
+
+  @media screen and (max-width: pxToEm($width))
     {block}
+
+
+/*
+ * Transform pixel values to REM/EM
+ */
+
+pxTo($px, $unit)
```
Comment (jonathanhudak) [inline src/styles/global/mixins.styl:23]: Let's just use `px` ;-)

http://html5weekly.us1.list-manage.com/track/click?u=0618f6a79d6bb9675f313ceb2&id=73725a8931&e=4df4fd938a

**Review by jonathanhudak — CHANGES_REQUESTED** [review]
Hey left some feedback. I have some local work I'll push up in the next few days as a PR.

File: src/routes/patterns/components/Alerts.js (hunk 1 above)
Comment (shiftsave) [reply src/routes/patterns/components/Alerts.js:11]: Super rough all around I'm gonna go ahead and trim this PR significantly! just wanted to get some starter files here!

File: src/routes/patterns/components/Alerts.js (hunk 2 above)
Comment (shiftsave) [reply src/routes/patterns/components/Alerts.js:37]: totally, gonna rework all of this, just wanted a starting point

File: src/styles/global/mixins.styl (hunk 4 above)
Comment (shiftsave) [reply src/styles/global/mixins.styl:23]: Ah the religion of em vs rem vs px, em-based mqs make more sense tho and rem keep it more consistent. This is not really a pain if we use a pxtorem function like the one I have in there. Check this other article out - https://zellwk.com/blog/media-query-units/

Comment (jonathanhudak) [reply src/styles/global/mixins.styl:23]: I'm down for em media queries and rem padding/margin font-size.

**Pull request merged**